zkevm-circuits/src/evm_circuit/op_execution/push.rs

```diff
@@ -0,0 +1,337 @@
+
+            // interpreter state transition constraints
+            let state_transition_constraints = vec![
+                state_next.global_counter.expr()
+                    - (state_curr.global_counter.expr() + 1.expr()),
+                state_next.program_counter.expr()
+                    - (state_curr.program_counter.expr()
+                        + 1.expr()
+                        + num_pushed.clone()),
+                state_next.stack_pointer.expr()
+                    - (state_curr.stack_pointer.expr() - 1.expr()),
+                state_next.gas_counter.expr()
+                    - (state_curr.gas_counter.expr() + GasCost::FASTEST.expr()),
+            ];
+
+            let PushSuccessAllocation {
+                case_selector,
+                word,
+                selectors,
+            } = &self.success;
+
+            let mut push_constraints = vec![];
+            for idx in 0..31 {
+                // selector can transit from 1 to 0 only once as [1, 1, 1, ...,
+                // 0, 0, 0]
+                if idx > 0 {
+                    let diff =
+                        selectors[idx - 1].expr() - selectors[idx].expr();
+                    push_constraints.push(diff.clone() * (1.expr() - diff));
+                }
+                // selectors needs to be 0 or 1
+                push_constraints.push(
+                    selectors[idx].expr() * (1.expr() - selectors[idx].expr()),
+                );
+                // word byte should be 0 when selector is 0
+                push_constraints.push(
+                    word.cells[idx].expr() * (1.expr() - selectors[idx].expr()),
+                );
+            }
+
+            let selectors_sum =
+                selectors.iter().fold(0.expr(), |sum, s| sum + s.expr());
+            push_constraints.push(selectors_sum - num_pushed);
+
+            let bus_mapping_lookups = [
+                // TODO: add 32 Bytecode lookups when supported
+                // selectors
+                //     .iter()
+                //     .zip(word.cells.iter())
+                //     .enumerate()
+                //     .map(|(idx, (selector, cell))| {
+                //         let OpExecutionState {
+                //             program_counter, ..
+                //         } = &state_curr;
+                //         Lookup::BusMappingLookup(BusMappingLookup::Bytecode {
+                //             hash: selector.expr() * hash,
+                //             index: selector.expr()
+                //                 * (program_counter.expr()
+                //                     + one.clone()
+                //                     + Expression::Constant(F::from_u64(
+                //                         idx as u64,
+                //                     ))),
+                //             value: cell.expr(),
+                //         })
+                //     })
+                //     .collect::<Vec<_>>(),
```

Comment:
Nitpick: move this to a new issue too. Avoid commit commented code.
```suggestion
```

Reply:
Fixed in [`26019ee`](https://github.com/appliedzkp/zkevm-circuits/pull/26/commits/26019ee70b8a7ee231d6faf4275722ef92105076)